dbms/src/Flash/Mpp/MPPTaskManager.cpp

```diff
@@ -183,37 +141,21 @@ std::pair<bool, String> MPPTaskManager::registerTask(MPPTaskPtr task)
     {
         mpp_query_map[task->id.start_ts]->task_map.emplace(task->id, task);
     }
-    task->manager = this;
+    task->registered = true;
     cv.notify_all();
     return {true, ""};
 }
 
-void MPPTaskManager::waitUntilQueryStartsAbort(UInt64 query_id)
+std::pair<bool, String> MPPTaskManager::unregisterTask(MPPTask * task)
```

Comment:
Seems this function only need `task.id`. Passing a `task` pointer is excessive.

Reply:
Updated.